app/src/main/java/nu/yona/app/YonaApplication.java

```diff
@@ -100,12 +102,27 @@ public void onCreate()
 		super.onCreate();
 		mContext = this;
 		eventChangeManager = new EventChangeManager();
-		sharedAppPreferences = eventChangeManager.getSharedPreference().getUserPreferences();
-		sharedUserPreferences = eventChangeManager.getSharedPreference().getAppPreferences();
+		validateAndInitializePreferences();
 		sharedAppDataState = eventChangeManager.getDataState();
 		Foreground.init(this);
 	}
 
+	private void validateAndInitializePreferences()
+	{
+		sharedAppPreferences = eventChangeManager.getSharedPreference().getAppPreferences();
+		sharedUserPreferences = eventChangeManager.getSharedPreference().getUserPreferences();
+		if (sharedUserPreferences.getString(AppConstant.SERVER_URL, null) != null)
+		{  // Need to exchange the preferences as corrective action, as a couple of previous builds had wrong(Exchanged Preferences) initializations.
+			nu.yona.app.utils.Logger.logi("Preferences", "Preferences Exchanged");
+			SharedPreferences tempSharedUserPrefs = YonaApplication.getAppContext().getSharedPreferences("TEMP_USER_PREF", Context.MODE_PRIVATE);
+			AppUtils.copySharedPreferences(sharedAppPreferences, tempSharedUserPrefs, true);
```

Comment:
You're always calling it with ``true``, right? If so, please omit the parameter and call the method ``moveSharedPreferences``, as you're practically moving them from one place to another.

Reply:
Done.